packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -583,14 +583,13 @@ class AndroidSdk {
     if (platform.isMacOS) {
       try {
         final String javaHomeOutput = processUtils.runSync(
-          <String>['/usr/libexec/java_home'],
+          <String>['/usr/libexec/java_home', '-v', '1.8'],
```

Comment:
At a glance, this does not seem very future proof. Should this not specify a version, and take the most recent available instead? /cc @blasten for thoughts.

Reply:
My initial thought too, but then I realized the error was due to the fact that the tooling was picking the latest version, and it fails whenever that latest version is not v1.8, hence my reason for this approach.

Reply:
do you remember which version `java_home` was defaulting too?

Reply:
For example, it defaults to 1.13 on my machine, which obviously do not work in flutter.
The first solution was uninstalling other versions of Java, then it would pick 1.8
Basically, it pick the latest version of java based on the output from `java_home`

Reply:
Thanks for the background information. One more question, did you ever get an error message other than the Flutter doctor failure? I wonder if there are related issues that can be associated with this fix.

Reply:
I did not get any other errors.
In fact, the original error was that "android-sdk" was not installed, after reading through similar errors from here and StackOverflow, I learned it was due to not having the right version of JDK (v1.8) installed.